### PR TITLE
[CLOUD-2545] Added clustering and draining

### PR DESCRIPTION
**Pull request by vbusch**
- amq-broker-71-statefulset-clustered-drain.yaml works with a drain controller to drain the messages
- renamed jolokia port, to hide the link in openshift

Signed-off-by: Vanessa Busch <vbusch@redhat.com>

Thanks for submitting your Pull Request!

Please make sure your PR meets the following requirements:

- [ ] Pull Request title is properly formatted: `[CLOUD-XYA] Subject`
- [ ] Pull Request contains link to the JIRA issue
- [ ] Pull Request contains description of the issue
- [ ] Pull Request does not include fixes for issues other than the main ticket
- [ ] Attached commits represent units of work and are properly formatted
- [ ] You have read and agreed to the Developer Certificate of Origin (DCO) (see `CONTRIBUTING.md`)
- [ ] Every commit contains `Signed-off-by: Your Name <yourname@example.com>` - use `git commit -s`

### CODE REVIEW

File: templates/amq-broker-71-statefulset-clustered-drain.yaml [hunk 1]
```diff
@@ -0,0 +1,334 @@
...
+  xpaas: 1.4.12
+message: A new messaging service has been created in your project. It will handle the protocol(s) "${AMQ_PROTOCOL}". The username/password for accessing the service is ${AMQ_USER}/${AMQ_PASSWORD}.
+metadata:
+  annotations:
+    description: Application template for JBoss AMQ brokers. These can be deployed as standalone or in a mesh. This template doesn't feature SSL support.
+    iconClass: icon-amq
+    openshift.io/display-name: JBoss AMQ Broker 7.1 (Ephemeral, no SSL)
+    openshift.io/provider-display-name: Red Hat, Inc.
+    tags: messaging,amq,jboss
+    version: 1.4.9
```
Comment (RoddieKieley) [inline templates/amq-broker-71-statefulset-clustered-drain.yaml]: @vbusch Not sure if you want to update this before the PR gets merged.

File: templates/amq-broker-71-statefulset.yaml [hunk 2]
```diff
@@ -0,0 +1,237 @@
+apiVersion: v1
+kind: Template
+labels:
+  template: amq-broker-71-statefulset
+  xpaas: 1.4.10
```
Comment (RoddieKieley) [inline templates/amq-broker-71-statefulset.yaml:5]: @vbusch Not sure if you want to update this before the PR gets merged.

File: templates/amq-broker-71-statefulset.yaml [hunk 3]
```diff
@@ -0,0 +1,237 @@
...
+message: A new messaging service has been created in your project. It will handle the protocol(s) "${AMQ_PROTOCOL}". The username/password for accessing the service is ${AMQ_USER}/${AMQ_PASSWORD}.
+metadata:
+  annotations:
+    description: Application template for JBoss AMQ brokers. These can be deployed as standalone or in a mesh. This template doesn't feature SSL support.
+    iconClass: icon-amq
+    openshift.io/display-name: JBoss AMQ Broker 7.1 (Ephemeral, no SSL)
+    openshift.io/provider-display-name: Red Hat, Inc.
+    tags: messaging,amq,jboss
+    version: 1.4.9
```
Comment (RoddieKieley) [inline templates/amq-broker-71-statefulset.yaml:14]: @vbusch Not sure if you want to update this before the PR gets merged.

File: templates/amq-broker-71-statefulset.yaml [hunk 4]
```diff
@@ -0,0 +1,237 @@
...
+            value: ${AMQ_QUEUES}
+          - name: AMQ_ADDRESSES
+            value: ${AMQ_ADDRESSES}
+          - name: AMQ_GLOBAL_MAX_SIZE
+            value: ${AMQ_GLOBAL_MAX_SIZE}
+          - name: AMQ_QUEUE_MEMORY_LIMIT
+            value: ${AMQ_QUEUE_MEMORY_LIMIT}
+          - name: AMQ_DATA_DIR
+            value: ${AMQ_DATA_DIR}
+          image: amq-broker71-openshift:1.0
```
Comment (RoddieKieley) [inline templates/amq-broker-71-statefulset.yaml:134]: @vbusch amq-broker-71-openshift:1.0 ?

File: templates/amq-broker-71-statefulset.yaml [hunk 5]
```diff
@@ -0,0 +1,237 @@
...
+          requests:
+            storage: 1Gi            
+    triggers:
+    - imageChangeParams:
+        automatic: true
+        containerNames:
+        - ${APPLICATION_NAME}-amq
+        from:
+          kind: ImageStreamTag
+          name: amq-broker71-openshift:1.0
```
Comment (RoddieKieley) [inline templates/amq-broker-71-statefulset.yaml:186]: @vbusch amq-broker-71-openshift:1.0 ?

File: modules/amq-launch/added/launch.sh [hunk 6]
```diff
@@ -57,57 +59,74 @@ function configureSSL() {
...
-    sed -i -ne "/<acceptors>/ {p; i $acceptors" -e ":a; n; /<\/acceptors>/ {p; b}; ba}; p" ${instanceDir}/etc/broker.xml
+    	IFS=',' read -a protocols <<< $(find_env "AMQ_TRANSPORTS" "openwire,amqp,stomp,mqtt,hornetq")
+    	connectionsAllowed=$(find_env "AMQ_MAX_CONNECTIONS" "1000")
+
+    	if [ "${#protocols[@]}" -ne "0" ]; then
+    		acceptors=""
+    		for protocol in ${protocols[@]}; do
+    		case "${protocol}" in
+        	"openwire")
+				acceptors="${acceptors}            <acceptor name=\"artemis-ssl\">tcp://${ACCEPTOR_IP}:61617?tcpSendBufferSize=1048576;tcpReceiveBufferSize=1048576;protocols=CORE,AMQP,STOMP,HORNETQ,MQTT,OPENWIRE;useEpoll=true;amqpCredits=1000;amqpLowCredits=300;connectionsAllowed=${connectionsAllowed};sslEnabled=true;keyStorePath=${keyStorePath};keyStorePassword=${keyStorePassword}</acceptor>\n"
```
Comment (dsimansk) [inline modules/amq-launch/added/launch.sh:75]: @vbusch does this change completely intentionally omit non-secured variant of `acceptors`?

File: templates/amq-broker-71-statefulset-clustered-drain.yaml (hunk 1 above)
Comment (RoddieKieley) [reply templates/amq-broker-71-statefulset-clustered-drain.yaml]: @vbusch also a couple of ip's in the 'image:'s at [L134](https://github.com/vbusch/jboss-amq-7-broker-openshift-image/blob/dca146ffa5310c12b0c26384388f18f04ebfd3f8/templates/amq-broker-71-statefulset-clustered-drain.yaml#L134) and [L204](https://github.com/vbusch/jboss-amq-7-broker-openshift-image/blob/dca146ffa5310c12b0c26384388f18f04ebfd3f8/templates/amq-broker-71-statefulset-clustered-drain.yaml#L204)

File: modules/amq-launch/added/launch.sh (hunk 6 above)
Comment (vbusch) [reply modules/amq-launch/added/launch.sh:75]: The launch scripts calls '/opt/amq/bin/artemis create', which populates the non-secure acceptors.  Afterwards, updateAcceptors adds the secure ones.

File: templates/amq-broker-71-statefulset-clustered-drain.yaml (hunk 1 above)
Comment (vbusch) [reply templates/amq-broker-71-statefulset-clustered-drain.yaml]: Updating it now

File: templates/amq-broker-71-statefulset-clustered.yaml [hunk 7]
```diff
@@ -0,0 +1,285 @@
...
+  xpaas: 1.4.12
+message: A new messaging service has been created in your project. It will handle the protocol(s) "${AMQ_PROTOCOL}". The username/password for accessing the service is ${AMQ_USER}/${AMQ_PASSWORD}.
+metadata:
+  annotations:
+    description: Application template for JBoss AMQ brokers. These can be deployed as standalone or in a mesh. This template doesn't feature SSL support.
+    iconClass: icon-amq
+    openshift.io/display-name: JBoss AMQ Broker 7.1 (Ephemeral, no SSL)
+    openshift.io/provider-display-name: Red Hat, Inc.
+    tags: messaging,amq,jboss
+    version: 1.4.9
```
Comment (RoddieKieley) [inline templates/amq-broker-71-statefulset-clustered.yaml:14]: @vbusch Not sure if you want to update this before the PR gets merged.

**Comment by vbusch** [comment]
Merging after hearing back from @RoddieKieley

**Comment by RoddieKieley** [comment]
@vbusch Nothing new in the comments, just hit submit on the review instead of cancel.

**Pull request merged**